hazelcast/serialization/predicate.py

```diff
@@ -204,6 +205,133 @@ def write_data(self, object_data_output):
+            output.write_object(anchor_value)
+
+    def next_page(self):
+        """
+        Sets page index to next page. If new index is out of range, the query results that this paging predicate will
+        retrieve will be an empty list.
+        :return (int) current page index
+        """
+        self.page += 1
+        return self.page
+
+    def previous_page(self):
+        """
+        If current page index is 0, this method does nothing. Otherwise, it sets page index to previous page.
+        :return (int) current page index
+        """
+        if self.page != 0:
+            self.page -= 1
+        return self.page
+
+    def set_page(self, page_no):
+        """
+        Sets page index to specified page_no.
+        If page_no is out of range, the query results that this paging predicate will retrieve will be an empty list.
+        :param page_no: (int) greater than or equal to 0.
+        :return (int) current page index
+        """
+        if page_no < 0:
+            raise ValueError('page_no should be positive or 0.')
+        self.page = page_no
+        return self.page
+
+    def set_iteration_type(self, iter_type):
+        """
+        :param iter_type: (ITERATION_TYPE)
+        """
+        self.iteration_type = iter_type
+
+    def set_anchor(self, nearest_page, anchor):
+        anchor_entry = (nearest_page, anchor)
+        anchor_count = len(self.anchor_list)
+        if nearest_page < anchor_count:
+            self.anchor_list[nearest_page] = anchor_entry
+        elif nearest_page == anchor_count:
+            self.anchor_list.append(anchor_entry)
+        else:
+            raise IndexError('Anchor index is not correct, expected: ' + str(self.page) + 'found: ' + str(anchor_count))
+
+    def reset(self):
+        self.iteration_type = ITERATION_TYPE.ENTRY
+        self.anchor_list.clear()
+        self.page = 0
+
+    def get_page(self):
+        return self.page
+
+    def get_page_size(self):
+        return self.page_size
+
+    def get_nearest_anchor_entry(self):
+        """
+        After each query, an anchor entry is set for that page.
+        For the next query user may set an arbitrary page.
+        For example: user queried first 5 pages which means first 5 anchor is available
+        if the next query is for the 10th page then the nearest anchor belongs to page 5
+        but if the next query is for the 3nd page then the nearest anchor belongs to page 2
+
+        :return nearest anchored entry for current page
+        """
+        anchor_count = len(self.anchor_list)
+        if self.page == 0 or anchor_count == 0:
+            return PagingPredicate.__NULL_ANCHOR
+        return self.anchor_list[self.page - 1] if self.page < anchor_count else self.anchor_list[anchor_count - 1]
+
+    def get_iteration_type(self):
+        return self.iteration_type
+
+    def get_comparator(self):
+        return self.comparator
+
+    def get_predicate(self):
```

Comment:
Unused method

Reply:
I took this out. But I have a question out of curiosity. If we leave the comparator field public, this allows the user to change the comparator function at any point after initializing the paging predicate, and I don't think we support that. Wouldn't it be safer to make the comparator field protected and to put a getter for it?

Reply:
I ended up making it protected and defining a property for it as you suggested below. Hope this works.

Reply:
I think you already have the answer but If you don't define a setter for the property, they became read-only